helpers.go

```diff
@@ -507,7 +507,7 @@ func repoConfigAsMount(config PackageRepositoryConfig, platformCfg *RepoPlatform
 
 	for name, repoConfig := range config.Config {
 		// each of these sources represent a repo config file
-		repoConfigSt, err := repoConfig.AsState(name, sOpt, append(opts, ProgressGroup("Importing repo config: "+name))...)
+		repoConfigSt, err := repoConfig.AsMount(name, sOpt, append(opts, ProgressGroup("Importing repo config: "+name))...)
```

Comment:
what's the difference between `AsMount` and `AsState`?

Reply:
I mentioned this in the second commit.
This is a minor optimization that avoids an extra copy operation.

The difference between `AsState` and `AsMount` is that `AsMount` expects that you are going to mount the returned state somewhere and can just use `llb.SourcePath` in that mount call rather than copying the data from the subpath into the root of a new `llb.State`

Basically, when using `AsMount` you need to create a mount that points at the requested subpath.
When using `AsState`, the state itself reflects that subpath but can require an extra copy (depending on the source type).

Reply:
Ah ok, I see it now:
```
This removes some extra copy operations required to use repo keys and
configs.
```

Ok yeah that makes sense as an optimization. Thanks for clarifying!